vaadin-button-flow-parent/vaadin-button-flow/src/main/java/com/vaadin/flow/component/button/GeneratedVaadinButton.java

```diff
@@ -110,32 +111,8 @@
 @NpmPackage(value = "@vaadin/vaadin-button", version = "23.1.0-rc3")
 @JsModule("@vaadin/button/src/vaadin-button.js")
 public abstract class GeneratedVaadinButton<R extends GeneratedVaadinButton<R>>
-        extends Component
-        implements HasStyle, ClickNotifier<R>, HasText, Focusable<R>, HasTheme {
-
-    /**
-     * Adds theme variants to the component.
-     *
-     * @param variants
-     *            theme variants to add
-     */
-    public void addThemeVariants(ButtonVariant... variants) {
-        getThemeNames()
-                .addAll(Stream.of(variants).map(ButtonVariant::getVariantName)
-                        .collect(Collectors.toList()));
-    }
-
-    /**
-     * Removes theme variants from the component.
-     *
-     * @param variants
-     *            theme variants to remove
-     */
-    public void removeThemeVariants(ButtonVariant... variants) {
-        getThemeNames().removeAll(
-                Stream.of(variants).map(ButtonVariant::getVariantName)
-                        .collect(Collectors.toList()));
-    }
+        extends Component implements HasStyle, ClickNotifier<R>, HasText,
+        Focusable<R>, HasTheme, HasThemeVariant<ButtonVariant> {
```

Comment:
For some reason all the interfaces are implemented for `GeneratedVaadinButton`, even though it's never reused. Decided to add this here as well just in case.